save-orchestrator-common/src/main/kotlin/com/saveourtool/save/orchestrator/service/AgentService.kt

```diff
@@ -104,32 +106,33 @@ class AgentService(
     /**
      * This method should be called when all agents are done and execution status can be updated and cleanup can be performed
      *
-     * @param containerId an ID of the agent from the execution, that will be checked.
+     * @param executionId an ID of the execution, that will be checked.
      */
     @Suppress("TOO_LONG_FUNCTION", "AVOID_NULL_CHECKS")
-    internal fun finalizeExecution(containerId: String) {
+    internal fun finalizeExecution(executionId: Long) {
         // Get a list of agents for this execution, if their statuses indicate that the execution can be terminated.
         // I.e., all agents must be stopped by this point in order to move further in shutdown logic.
-        getFinishedOrStoppedAgentsForSameExecution(containerId)
-            .filter { (_, finishedContainerIds) -> finishedContainerIds.isNotEmpty() }
-            .flatMap { (_, _) ->
+        getFinishedOrStoppedAgentsByExecutionId(executionId)
+            .filter { finishedContainerIds -> finishedContainerIds.isNotEmpty() }
+            .flatMap {
                 // need to retry after some time, because for other agents BUSY state might have not been written completely
-                log.debug("Waiting for ${configProperties.shutdown.checksIntervalMillis} ms to repeat `getAgentsAwaitingStop` call for containerId=$containerId")
+                log.debug("Waiting for ${configProperties.shutdown.checksIntervalMillis} ms to repeat `getAgentsAwaitingStop` call for execution=$executionId")
                 Mono.delay(Duration.ofMillis(configProperties.shutdown.checksIntervalMillis)).then(
-                    getFinishedOrStoppedAgentsForSameExecution(containerId)
+                    getFinishedOrStoppedAgentsByExecutionId(executionId)
                 )
             }
-            .filter { (_, finishedContainerIds) -> finishedContainerIds.isNotEmpty() }
-            .flatMap { (executionId, finishedContainerIds) ->
+            .filter { finishedContainerIds -> finishedContainerIds.isNotEmpty() }
+            .flatMap { finishedContainerIds ->
                 log.info { "For execution id=$executionId all agents have completed their lifecycle" }
                 markExecutionBasedOnAgentStates(executionId, finishedContainerIds)
-                    .thenReturn(
+                    .then(Mono.fromCallable {
+                        orchestratorAgentStatusService.deleteAllByExecutionId(executionId)
```

Comment:
May be rename this method, thus to point that we actually clear our collections and not removing containers from `db` or `docker`? As I remember, now `cleanup `is responsible for this

Reply:
actually, I did it on purposes -- `HeartBeatInspector` with info about containers mimics\repeats `AgentStatusRepository` from backend, but now it's in-memory.

The class and field has name with prefix `orchestrator` to highlight that it's on orchestrator level

Reply:
renamed the class and field name